### PR TITLE
Final fixes for the NGSIv2 release candidate May 2016

### PR DESCRIPTION
**Pull request by fgalan**
This PR finished the work to prepare the NGSIv2 specification release candidate May 2016.

Summary of changes:

* Rewritten "Status" section (with the Status corresponding to Release Candidates)
* Added section on "Reference Implementations"
* Removed references to "role" concept
* Removed empty placeholders "Terminology" subsections 
* Removed Context Element section (as it has not been yet decided if this concetp will be part of the final version of the API)
* Removed "pending discussion" notes.
* Removed "registration_url" removed from GET /v2 operation (as context management availability as not been yet defined in NGSIv2 specification)
* Removed "# Group Registrations" (as context management availability as not been yet defined in NGSIv2 specification)
* Removed "scopes" in POST /v2/op/query (as scopes has not been yet crearly defined)
* Removed POST /v2/op/register and POST /v2/op/discover (as context management availability as not been yet defined in NGSIv2 specification)
* Removed "# Group OMA-NGSI Operations" (only empty placeholders there)
* Removed references (and examples) to default entity/attribute/netadata type to "none" (this requires more disucssion, see https://github.com/telefonicaid/fiware-orion/issues/2223)
* Removed affectedItem field in "Error Responses" (not clear enougth)

Note that alghough many sections has been removed in this document, they remain in the latest version of the specification (fiware-ngsiv2-reference.apib file) so discussion can continue on them and, eventually, be added to next release candidates.

Merging deadline: **June 7th EoB.**

### CODE REVIEW

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 1]
```diff
@@ -32,16 +32,20 @@ Sergio García Gómez (Telefónica I+D), Martin Bauer (NEC).
...
+  of outgoing discussions. Backward compatibility will be taken into account in this case, trying
+  to minimize the impact on existing software implementations. In particular, only completely
+  justified changes impacting backward compatibility will be allowed and "matter of taste"
+  changes will not be allowed.
 
-* https://github.com/telefonicaid/fiware-orion/issues/1022
-* https://github.com/telefonicaid/fiware-orion/issues/1034
-* https://github.com/telefonicaid/fiware-orion/issues/1035
+The work in progress draft of the specification can be found at
+[the following link](https://telefonicaid.github.io/fiware-orion/api/v2/latest).
```
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Once we have the public URL for the detailed releases notes document a link could be added here.

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 2]
```diff
@@ -32,16 +32,20 @@ Sergio García Gómez (Telefónica I+D), Martin Bauer (NEC).
...
-https://lists.fiware.org/private/fiware-ngsi/ (list subscription required).
+This specification is a release candidate of the NGSIv2 API specification. By "release candidate"
+we mean that the specification is quite stable, but changes may occur with regard to new release
+candidates or the final version. In particular changes may be of two types:
 
-In addition, note that a list of currently open discussions is available at
+* Extensions to the functionality currently specified by this document. Note that in this case
+  there isn't any risk of breaking backward compatibility on existing software implementations.
+* Slight modifications in the functionality currently specified by this document, as a consequence
+  of outgoing discussions. Backward compatibility will be taken into account in this case, trying
```
Comment (kzangeli) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: outgoing ... ongoing?

**Comment by kzangeli** [comment]
LGTM, except the doubt about 'outgoing/ongoing' ...

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 2 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: You are right. Righ one is 'ongoing'.

Fixed in 7d94c7add182b0a378778c1fecb68c9eb2f7bdc5

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 3]
```diff
@@ -32,16 +32,20 @@ Sergio García Gómez (Telefónica I+D), Martin Bauer (NEC).
   
 ## Status
 
-This is a work in progress and is changing on a daily basis.
-Please send your comments to fiware-ngsi@lists.fiware.org.
-You can trace the discussions checking the archives of the mailing list:
-https://lists.fiware.org/private/fiware-ngsi/ (list subscription required).
+This specification is a release candidate of the NGSIv2 API specification. By "release candidate"
+we mean that the specification is quite stable, but changes may occur with regard to new release
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: I would rephrase this as "Only Minor Changes are expected to the API described by this document". Nonetheless new functionalities will be added, particularly JSON-LD and NGSI9 (context registration) support.

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 4]
```diff
@@ -32,16 +32,20 @@ Sergio García Gómez (Telefónica I+D), Martin Bauer (NEC).
...
+candidates or the final version. In particular changes may be of two types:
 
-In addition, note that a list of currently open discussions is available at
+* Extensions to the functionality currently specified by this document. Note that in this case
+  there isn't any risk of breaking backward compatibility on existing software implementations.
+* Slight modifications in the functionality currently specified by this document, as a consequence
+  of ongoing discussions. Backward compatibility will be taken into account in this case, trying
+  to minimize the impact on existing software implementations. In particular, only completely
+  justified changes impacting backward compatibility will be allowed and "matter of taste"
+  changes will not be allowed.
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: I don't know if it is really needed ... to be explain this here

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 5]
```diff
@@ -32,16 +32,20 @@ Sergio García Gómez (Telefónica I+D), Martin Bauer (NEC).
   
 ## Status
 
-This is a work in progress and is changing on a daily basis.
-Please send your comments to fiware-ngsi@lists.fiware.org.
-You can trace the discussions checking the archives of the mailing list:
-https://lists.fiware.org/private/fiware-ngsi/ (list subscription required).
+This specification is a release candidate of the NGSIv2 API specification. By "release candidate"
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: is the May 2016 release candidate

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 6]
```diff
@@ -407,8 +345,6 @@ If present, the error payload is a JSON object including the following fields:
 
 + `error` (required, string): a textual description of the error.
 + `description` (optional, string): additional information about the error.
-+ `affectedItems` (optional, array[string]): a list of elements affected by the error.
-  Depending on the operation, it may refer to entities, registrations or subscriptions.
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: why are we removing this?

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 7]
```diff
@@ -459,9 +395,6 @@ Thus, implementations may raise errors when spatial index limits are exceeded.
 The recommended HTTP status code for those situations is ``413``, *Request entity too large*, and
 the reported error on the response payload must be ``NoResourcesAvailable``.
 
-*Note: At the time of writing it is still under discussion whether supporting both syntaxes is
-really needed* 
-
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: why are we removing this?

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 8]
```diff
@@ -297,18 +241,12 @@ Some operations use partial representation of entities:
 
 * `id` and `type` are not allowed in update operations, as they are immutable properties.
 
-* In requests where entity `type` is allowed, it may be omitted. When omitted in entity
-  creation operations, the default string value `none` is used for the type.
-
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Thing?

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 9]
```diff
@@ -1848,188 +1779,6 @@ Response:
 
 + Response 204
 
-# Group Registrations
-
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: I would leave here a note indicating that this is a group of operations not part of this release but ongoing work.

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 1 above)
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: then it should point to fiware.github.io ...

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 10]
```diff
@@ -62,6 +66,11 @@ NGSI version 2 uses camel case (camelCase) syntax for naming properties and rela
 by the API. When  referring to URIs, as part of HATEOAS patterns, and to mark them appropriately,
 the suffix `_url` is added.
 
+## Reference Implementations
+
+* NGISv2 servers
+  * [Orion Context Broker](http://catalogue.fiware.org/enablers/publishsubscribe-context-broker-orion-context-broker) - [Implementation Notes](https://fiware-orion.readthedocs.io/en/develop/user/ngsiv2_implementation_notes/index.html)
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: nit line too long.

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib [hunk 11]
```diff
@@ -2186,200 +1925,3 @@ Response code:
...
-              "attributes": [
-                "speed"
-              ]
-            },
-            "callback": "http://localhost:5678",
-            "duration": "PT1M"
-          }
-        ]
-
-# Group OMA-NGSI Operations
```
Comment (jmcanterafonseca) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: I would leave a note here indicating that these operations are available under the /v1/* resource.

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 10 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: URL makes this line very long and I'm a bit unsure how breaking this into two lines could affect to the bullet list indentation.

Thus, this time, I prefer to leave it long :)

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 1 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Good poing... overall replacement to fiware.github.io would be done.

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 8 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: It seems so... but not sure yet. It depends on how we solve https://github.com/telefonicaid/fiware-orion/issues/2223.

Being conservative, it is better not saying anything about this in this release candidate (next ones will do).

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 6 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: `affectedItems` has been actually used in Orion reference implementation. In addition, some partners has raised concerns about this field.

Thus, being conservative, it is better not saying anything about this in this release candidate (next ones will do).

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 7 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Release Candidate text shouldn't include explicit references to open dicussions. That references to open discussions are fine, but within the context of the WIP version of the speficiaiton (i.e. the one that will remain in "/latest").

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 9 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Release Candidate text shouldn't include explicit references to open dicussions. That references to open discussions are fine, but within the context of the WIP version of the speficiaiton (i.e. the one that will remain in "/latest").

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 11 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Release Candidate text shouldn't include explicit references to open dicussions. That references to open discussions are fine, but within the context of the WIP version of the speficiaiton (i.e. the one that will remain in "/latest").

Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Well... actually in this case ragher than reference to an open dicussion is a kind of cross-version comment. It would be a bit weird...

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 4 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: I think is a good idea to provide information to potential API users about the kind of changes they can expect over this release candidate.

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 9 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: However, that note will be perfect to be included in the Release Notes associated to thir release candidate.

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 10 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: NTC

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 11 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: However, that note will be perfect to be included in the Release Notes associated to thir release candidate.

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 1 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Fixed in bce7b1bf98f3909bee70d74b4f422f0bf60f72bc

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 5 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Fixed in bce7b1bf98f3909bee70d74b4f422f0bf60f72bc

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 3 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: Fixed in 6909f2d

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 4 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: NTC

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 6 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: NTC

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 7 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: NTC

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 8 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: NTC

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 9 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: NTC

File: doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib (hunk 11 above)
Comment (fgalan) [inline doc/apiary/v2/fiware-ngsiv2-rc-2016_05_31.apib]: NTC
